sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -206,15 +207,35 @@ def _infer_event_timestamp_from_dataframe(entity_df: pandas.DataFrame) -> str:
 
 
 class BigQueryRetrievalJob(RetrievalJob):
-    def __init__(self, query, client):
+    def __init__(self, query, client, config):
         self.query = query
         self.client = client
+        self.config = config
 
     def to_df(self):
         # TODO: Ideally only start this job when the user runs "get_historical_features", not when they run to_df()
         df = self.client.query(self.query).to_dataframe(create_bqstorage_client=True)
         return df
 
+    def to_bigquery(self, dry_run=False) -> str:
```

Comment:
> def to_bigquery(self, dry_run=False) -> str:

are we then committing to the idea that execution for all job types only starts when the RetrievalJob object has its `to_X()` methods triggered? (which is different from what I was expecting here https://github.com/feast-dev/feast/pull/1634/files#diff-14f38fcbdc2240bb60b9f558d133b174f5b6059cfcd959eef8b9c49bb5be8838R216, but also a viable approach)

Reply:
Also, should the dry run configuration not be independent of the destination? I can imagine that `to_df()` and `to_bigquery()` could use the same underlying means of execution or at least configuration.

Reply:
> are we then committing to the idea that execution for all job types only starts when the RetrievalJob object has its to_X() methods triggered

I feel that this is the best practice IMO

Reply:
> Also, should the dry run configuration not be independent of the destination?

Unsure about its impact on 
```python
print("This query will process {} bytes.".format(bq_job.total_bytes_processed))
```